pkg/ipam/funcs.go

```diff
@@ -15,12 +13,45 @@ type AssignIPDetails struct {
 	NetName     string
 	SubNet      *networkv1.Subnet
 	Reservelist *networkv1.ReservationList
-	FixedIP     net.IP
+	FixedIP     netip.Addr
+	// internal use only
+	excluded map[netip.Addr]bool
+	reserved map[netip.Addr]string
+}
+
+func (a *AssignIPDetails) buildExcluded() {
+	if a.excluded != nil {
+		return
+	}
+	a.excluded = make(map[netip.Addr]bool, len(a.SubNet.ExcludeAddresses))
+	for _, ipStr := range a.SubNet.ExcludeAddresses {
+		if ip, err := netip.ParseAddr(ipStr); err == nil {
```

Comment:
We validate the ExcludeAddresses via the webhook, but not using netip. https://github.com/openstack-k8s-operators/infra-operator/blob/main/apis/network/v1beta1/netconfig_webhook.go#L345 
Should there be a validation here, too? or is it ok to just ignore a ParseAddr error?

Reply:
good call. fixed